.github/gradle/gradle.gradle

```diff
@@ -1,41 +1,131 @@
     return
 }
 
-static def outputUnitTest(Project project, Task test) {
-    project.getLogger().lifecycle("<< TEST >>${test.path}")
+static Map<String, String> removeCommonPrefix(List<String> list) {
+    if (list.size() == 0) {
+        return new HashMap<String, String>()
+    }
+
+    def result = new HashMap<String, String>()
+
+    if (list.size() == 1) {
+        System.out.println("Only one item in the list: ${list}")
+        def sections = list[0].split("\\.")
+        result.put(list[0], sections.last())
+        return result
+    }
+
+    Collections.sort(list, { String a, String b -> a.length() - b.length() })
+    def max = list[0].length()
+    def prefix = ""
+    for(int index = 0; index < max; index++) {
+        def currentChar = list[0][index]
+        for (String item : list) {
+            if (item[index] != currentChar) {
+                for (final def entry in list) {
+                    result.put(entry, entry.substring(prefix.length()))
+                }
+                return result
+            }
+        }
+
+        prefix += currentChar
+    }
+
+    throw new RuntimeException("Could not remove common prefix: ${list}")
+}
+
+static String createDisplayNameSuffix(String path, String filter) {
+    //The path starts with a : and then the project name:
+    path = path.substring(path.indexOf(":") + 1)
+
+    //Now split the path into parts
+    def parts = path.split(":")
+    def capitalizedParts = parts.collect { it.capitalize() }
+
+    //Combine the parts with ->
+    path = capitalizedParts.join(" -> ")
+
+    if (filter === null) {
+        return path
+    }
+
+    //Next the filter has its common prefix stripped, is however still in domain name form
+    def filterParts = filter.split("\\.")
+    def capitalizedFilterParts = filterParts.collect { it.capitalize() }
+    filter = capitalizedFilterParts.join("/")
+
+    return "${path} (${filter})"
+}
+
+static List<TestRun> createTestRuns(Task it, List<String> testClasses) {
+    def testRuns = new ArrayList<TestRun>()
+    if (testClasses.size() == 0) {
+        testRuns.add(new TestRun(displayName: "Test - ${createDisplayNameSuffix(it.path, null)}", path: it.path, filter: null))
+        return testRuns
+    }
+
+    def testClassesWithoutCommonPrefix = removeCommonPrefix(testClasses)
+    testClassesWithoutCommonPrefix.forEach { className, displayName ->
+        testRuns.add(new TestRun(displayName: "Test - ${createDisplayNameSuffix(it.path, displayName)}", path: it.path, filter: className))
+    }
+
+    return testRuns
 }
 
-static def outputClassTest(Project project, Task test, String className) {
-    project.getLogger().lifecycle("<< TEST >>${test.path} --tests \"${className}\"")
+@CompileStatic
+class TestRun {
+    String displayName
+    String path
+    String filter
+
+    String getDisplayName() {
+        return displayName
+    }
+
+    String getPath() {
+        return path
+    }
+
+    String getFilter() {
+        return filter
+    }
```

Comment:
This is groovy, you already have getters generated, which are used by default.

Reply:
Sadly that does not actually suffice for the JsonBuilder included in groovy
I tested it without cause I thought it would do it, but then it does not include the properties in the json what so ever.

Reply:
Seems to work now...
Not sure what I fucked up when I thought I needed to add them